.github/workflows/codeql.yml

```
@@ -190,6 +162,32 @@ jobs:
 
         echo "Go environment prepared for autobuild"
 
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v4 # v4.0.0
```

```suggestion
      uses: github/codeql-action/init@cf1bb45a277cb3c205638b2cd5c984db1c46a412 # v4.31.7
```
Just a nit to pin the version. This way is more align with the rest workflows, not braking hopefully seemingly randomly because of an update. Or if we would like to keep the `v4` tag, then plase remove the comment next to it as not correct.